### DISCUSSION TITLE
Can you elaborate on what the upcoming pro features on?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [johnlindquist/kit#1405] [General]
It says this in the app
> Upcoming Pro Feature
> Sync Scripts to GitHub Repo
> Run Script Remotely as GitHub Actions
> Advanced Widgets
> Screenshots
> Screen Recording
> Audio Recording
> Webcam Capture
> Desktop Color Picker
> Measure Tool

What exactly are the following:
- Advanced Widgets
- Screenshots
- Screen & Audio Recording
- Webcam capture: What exactly would be done with this?
- Desktop Color picker: How is it different from existing color picker apps, especially on macOS?
- Measure Tool

**Comment by johnlindquist** [comment]
## Advanced Widgets

Spinning up a custom React/Svelte/etc server and hosting it inside a widget with a backend so you have complete control over the UI.

## Screenshots

A built-in `await screenshot()` tool that uses a standard screenshot UI for selecting an area of the screen

## Screen/Audio Recording

Same as screenshot, but for recording your screen

## Webcam capture

We currently have `await webcam()` for capturing an image. This would be for recording

## Color picker

The different being if would be scriptable

```js
let color = await colorPicker()
// do whatever you want with the color
```

## Measure

Get the width/height of anything on your screen.
```js
let rectangle = await measure()
```
